### ISSUE TITLE
[Bug]: aws_elb_service_account data source fails with Error: Unknown region ("me-central-1") 

### ISSUE THREAD

**Issue by mcricker** [hashicorp/terraform-provider-aws#29322]
### Terraform Core Version

Terraform v0.14.11

### AWS Provider Version

 v4.53.0

### Affected Resource(s)

aws_elb_service_account

### Expected Behavior

aws_elb_service_account data source should work for the new region me-central-1

### Actual Behavior

when doing a terraform apply it fails at this block
data "aws_elb_service_account" "main" {
}
with 
Error: Unknown region ("me-central-1") 

### Relevant Error/Panic Output Snippet

```shell
Error: Unknown region ("me-central-1")
```

### Terraform Configuration Files

data "aws_elb_service_account" "main" {
}

### Steps to Reproduce

try to set up the aws_elb_service_account data source in me-central-1

### References

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/elb_service_account#region

### Would you like to implement a fix?

None

**Comment by ewbankkit** [comment]
Relates: https://github.com/hashicorp/terraform-provider-aws/issues/27874.

**Comment by ewbankkit** [comment]
@mcricker Thanks for raising this issue.

For the newer Regions AWS document that service principals should be used instead of account IDs:

* https://docs.aws.amazon.com/redshift/latest/mgmt/db-auditing.html#db-auditing-bucket-permissions
* https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/enable-access-logs.html#attach-bucket-policy
* https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-access-logging.html#attach-bucket-policy

The maintainers will discuss what we do with such Regions in these data sources.

**Comment by mcricker** [comment]
Can service principals be used in the same way for older regions as well (ie not need specific terraform for specific regions)

**Comment by ewbankkit** [comment]
I have noticed that we don't return a Regions' opt-in status from either the `aws_region` or `aws_regions` data sources.
You could use a filter on the `aws_regions` data source and then the [Terraform `contains` function](https://developer.hashicorp.com/terraform/language/functions/contains) on the resulting list of `names`.

**Comment by mcricker** [comment]
> I have noticed that we don't return a Region's _opt-in status_ from either the `aws_region` or `aws_regions` data sources. You could use a filter on the `aws_regions` data source and then the [Terraform `contains` function](https://developer.hashicorp.com/terraform/language/functions/contains) on the resulting list of `names`.

I don't think that this will actually work as a way to differentiate these regions because there are older opt-in regions that do not have the same behaviour (eg me-south-1)

**Comment by mcricker** [comment]
In addition to the above issue, I am wondering if there is a way to check if lambdas in the region support arm64 architecture? Since this is also an issue with me-central-1

**Comment by xeonic-ant** [comment]
Hi. Problem is really serious, we can not use terraform for new regions
https://github.com/hashicorp/terraform-provider-aws/issues/28909

**Comment by chinaPancake** [comment]
I See there is nothing new to the topic? @ewbankkit

**Comment by jacoor** [comment]
Could use help here... still having the issue
